src/main/java/com/labsynch/labseer/domain/AnalysisGroupValue.java

```diff
@@ -1970,4 +1970,33 @@ public void setLsState(AnalysisGroupState lsState) {
 	public AnalysisGroupValue() {
 		super();
 	}
+
+    public static TypedQuery<AnalysisGroupValue> findAnalysisGroupValuesByLsKindEqualsAndCodeValueEquals(
```

Comment:
Didn't want to spend the time to try and abstract this query so you'll see this repeated on many different levels (subject, treatment, experiment..etc.)